lib/pinecone.ex

```diff
@@ -306,6 +306,34 @@ defmodule Pinecone do
     delete({:vectors, "#{name}-#{project_name}"}, "vectors/delete", opts[:config], params: params)
   end
 
+  @doc """
+  Deletes all vectors from the given Pinecone index that match the given metadata filter.
```

Comment:
Maybe not a great reason: the docs say the `ids`, `deleteAll`, and `filter` params are mutually exclusive, so I wanted to avoid making an interface that appears to accept combinations of inputs that are in fact inconsistent or invalid. I guess it wouldn't make it too confusing to have it be a part of `delete_all_vectors/2` though

Reply:
I made that change